numba/cuda/compiler.py

```diff
@@ -74,6 +71,50 @@ def compile_kernel(pyfunc, args, link, debug=False, inline=False,
     return cukern
 
 
+@global_compiler_lock
+def compile_ptx(pyfunc, args, debug=False, fastmath=False, cc=None, opt=3):
+    """Compile a Python function to PTX for the given arguments.
+
+    :param pyfunc: The Python function to compile.
+    :param args: A tuple of argument types to compile for.
+    :param debug: Whether to include debug info in the generated PTX.
+    :type debug: bool
+    :param fastmath: Whether to enable fast math flags (ftz=1, prec_sqrt=0,
+                     prec_div=, and fma=1)
+    :type fastmath: bool
+    :param cc: Compute capability to compile for, as a tuple ``(MAJOR, MINOR)``.
+               Defaults to ``(5, 0)``.
```

Comment:
Could you default to the current device's CC if `cc=None`?

Reply:
I'm trying to avoid `compile_ptx` having anything to do with devices - I picked 5.0 because I thought it was the lowest non-deprecated compute capability... However upon double-checking the release notes for 10.2 (https://docs.nvidia.com/cuda/cuda-toolkit-release-notes/index.html#deprecated-features) I see that is deprecated too, so I should probably use 5.2 instead.

Reply:
Thanks for the pointer, Graham. Wasn't aware of the deprecation. Yeah I did not realize device independency is done here, so it makes sense to pick one.

But, I find the release note rather confusing -- a few lines below it says 5.2 is also deprecated:
> Support for Kepler and Maxwell architectures (compute capability sm_35 through sm_52) is deprecated.

My impression from working with half-precision is that `sm_53` is the minimum requirement -- can you confirm?

Reply:
I think the release notes are a bit confusing, but from my reading of it, up to `sm_50` is deprecated in all of CUDA. However, for the CUDA libraries (cuBLAS, etc.) `sm_52` is also deprecated. I don't know anything about half-precision specifically.

As PTX is not specific to CUDA libraries, but (probably) falls under "General CUDA", I'm inclined to leave the default at `(5, 2)` - it seems that 5.3 is only implemented in Jetson, Tegra, and DRIVE, which are probably not the majority of use cases for Numba - if I were to pick a higher default than `(5, 2)` it would feel more appropriate to use `(6, 0)` instead.

Reply:
I see, yeah l think it's a good point. Thanks for clarifying.